development/build/transforms/code-fencing.js

```diff
@@ -0,0 +1,288 @@
+const path = require('path');
+const through = require('through2');
+const { PassThrough } = require('readable-stream');
+const { BuildTypes } = require('../utils');
+
+module.exports = {
+  createCodeFencingTransform,
+  removeFencedCode,
+};
+
+/*
+///: BEGIN:ONLY_INCLUDE_IN(flask)
+AssetsController: this.assetsController.store,
+PluginController: this.pluginController,
+///: END:ONLY_INCLUDE_IN
+ */
+
+/**
+ *
+ * @param {string} buildType - The type of the current build.
+ */
+function createCodeFencingTransform(buildType) {
+  if (!(buildType in BuildTypes)) {
+    throw new Error(
+      `Metamask build: Code fencing transform received unrecognized build type: "${buildType}"`,
+    );
+  }
+
+  // The transform is function that receives a file as an argument, and returns
+  // duplex stream that applies the transform logic by operating on the file
+  // contents as a buffer with a particular encoding (in our case always utf8).
+  // "next" is called at the end of the transform operation to inform browserify
+  // that it can continue to the next transform.
+  return function (fileName) {
+    if (!['.js', '.cjs', '.mjs'].includes(path.extname(fileName))) {
+      return new PassThrough();
+    }
+
+    return through(function (fileBuffer, _encoding, next) {
+      this.push(
+        removeFencedCode(fileName, buildType, fileBuffer.toString('utf8')),
```

Comment:
this makes assumptions on chunk contents. i reccomend operating on the whole file instead of chunks. as seen in the example i gave @rekmarks

Reply:
eg chunking barrier could happen in the middle of a fence identifier